js/repl/Repl.js

```diff
@@ -199,9 +200,22 @@ class Repl extends React.Component {
     );
   }
 
-  async _setupBabel() {
+  async _setupBabel(defaultPresets) {
     const babelState = await loadBabel(this.state.babel, this._workerApi);
-    this.setState(babelState);
+
+    // Filter the list of preloaded presets with those available with
+    // the loaded version of @babel/standalone.
+    const availablePresetConfigs = babelState.isLoaded
+      ? presetPluginConfigs.filter(
+          ({ label, isPreLoaded }) =>
+            !isPreLoaded || babelState.availablePresets.indexOf(label) > -1
```

Comment:
I'm just glancing at this PR while I'm waiting on a build to complete, so I'm probably misreading something silly, but this line doesn't make sense to me:
```js
!isPreLoaded || babelState.availablePresets.indexOf(label) > -1
```

If we're explicitly validating that the preset is available (loaded) then why are we even checking `isPreLoaded` (let alone treating `!isPreLoaded` the same as one that's actually been verified and loaded)?

Reply:
Originally, I actually replaced `presetPluginConfigs` with one generated from `babelState.availablePresets` but decided (perhaps foolishly) that maybe it was better to preserve the ability to have presets listed in the config that weren't preloaded and would get parsed later in `_checkForUnloadedPlugins`.

At least thats how I've been interpreting the intent of that bool for the entries in `presetPluginConfigs`.

Reply:
If I recall correctly, I actually added the bool thinking that we might want to add more presets eventually that required runtime loading (similar to how we load eg Prettier). Maybe we should just ditch it though, since they're all true now.

Either way, it seems odd to me that we filter this way. 😄  Seems like the `isPreLoaded` isn't important if we're explicitly checking the plugins that standalone Babel's reporting it has.

Reply:
I guess I was thinking that the filtered list should include the `isPreLoaded` presets that _are_ available via babel/standalone and any additional non preloaded presets to pass down to the next step in the process (at that point it can eventually runtime load stuff, etc).

Maybe we just simplify it altogether and generate `presetPluginConfigs` based on available presets from babel/standalone and tackle runtime loading differently (similar to preset-env).

Reply:
I think that would be a nice simplifying change for the time being 😄